aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,530 @@
-
-### What is the difference between a host and a server?
-
-The host is responsible for application startup and lifetime management. The server is responsible for accepting HTTP requests. Part of the host's responsibility includes ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-## Setting up a Host
+Create a host using an instance of [IWebHost](/dotnet/api/microsoft.aspnetcore.hosting.iwebhost). This is typically performed in your app's entry point, the `Main` method. In the project templates, `Main` is located in *Program.cs*. A typical *Program.cs* calls [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder) to build the host:
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+[!code-csharp[Main](../common/samples/WebApplication1DotNetCore2.0App/Program.cs?name=snippet_Main)]
 
-You create a host using an instance of `WebHostBuilder`. This is typically done in your app's entry point: `public static void Main` (which in the project templates is located in a *Program.cs* file). A typical *Program.cs*, shown below, demonstrates how to use a `WebHostBuilder` to build a host.
+`CreateDefaultBuilder` performs the following tasks:
 
-[!code-csharp[Main](../common/samples/WebApplication1/Program.cs?highlight=14,15,16,17,18,19,20,21)]
+* Configures [Kestrel](servers/kestrel.md) as the web server.
+* Sets the content root to [Directory.GetCurrentDirectory](/dotnet/api/system.io.directory.getcurrentdirectory).
+* Loads available configuration from:
+  * *appsettings.json*.
+  * *appsettings.{Environment}.json*.
+  * [User secrets](xref:security/app-secrets) when the app runs in the `Development` environment.
+  * Environment variables.
+  * Command-line arguments.
+* Configures [logging](xref:fundamentals/logging) for console and debug output with [log filtering](xref:fundamentals/logging#log-filtering) rules specified in a Logging configuration section of an *appsettings.json* or *appsettings.{Environment}.json* file.
+* Enables IIS integration by configuring the base path and port the server should listen on when using the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module). The module creates a reverse-proxy between Kestrel and IIS. Also configures the app to [capture startup errors](#capture-startup-errors).
+* Enables the [developer exception page](xref:fundamentals/error-handling#the-developer-exception-page) when the app runs in the `Development` environment.
```

Comment:
That's not in CreateDefaultBuilder: https://github.com/aspnet/MetaPackages/blob/586d641934e997045ed98b1978684a7b0c607b39/src/Microsoft.AspNetCore/WebHost.cs#L152. Isn't that in Startup in the template?

Reply:
It was added on this commit: https://github.com/aspnet/Docs/commit/424127e1ba0cc97cd6322c0613b6231c97c1b399#diff-c3121d49e1aa157dca834774d1650220R68

@tdykstra Just a typo?

Reply:
See line 148 Apparently I got it from the comments and didn't validate that the code was actually doing it.
https://github.com/aspnet/MetaPackages/blob/586d641934e997045ed98b1978684a7b0c607b39/src/Microsoft.AspNetCore/WebHost.cs#L148

Reply:
:smile: Ah. Thanks. I'll put in a quick ~patch there~. It's already done on the comments. I'll fix it here.

Reply:
Yup the xml docs were updated in https://github.com/aspnet/MetaPackages/issues/223.